# Data Cleaning WorldCupMatches Dataset

In [60]:
## Import libraries
import pandas as pd 
import numpy as np 
import matplotlib.pyplot as plt 
import seaborn as sns 
%matplotlib inline

pd.set_option('display.max_columns',None)

In [61]:
### Import Dataset 

Data = pd.read_csv('/Users/inazawaelectronics/Desktop/Data-Analyst-Intern-Project/FIFA WORLD CUP ANALYSIS PROJECT/Data/WorldCupMatches.csv')
Data.head()

Year              Datetime    Stage         Stadium         City  \
0  1930.0  13 Jul 1930 - 15:00   Group 1         Pocitos  Montevideo    
1  1930.0  13 Jul 1930 - 15:00   Group 4  Parque Central  Montevideo    
2  1930.0  14 Jul 1930 - 12:45   Group 2  Parque Central  Montevideo    
3  1930.0  14 Jul 1930 - 14:50   Group 3         Pocitos  Montevideo    
4  1930.0  15 Jul 1930 - 16:00   Group 1  Parque Central  Montevideo    

  Home Team Name  Home Team Goals  Away Team Goals Away Team Name  \
0         France              4.0              1.0         Mexico   
1            USA              3.0              0.0        Belgium   
2     Yugoslavia              2.0              1.0         Brazil   
3        Romania              3.0              1.0           Peru   
4      Argentina              1.0              0.0         France   

  Win conditions  Attendance  Half-time Home Goals  Half-time Away Goals  \
0                     4444.0                   3.0                   0.0   
1                    18346.0                   2.0                   0.0   
2                    24059.0                   2.0                   0.0   
3                     2549.0                   1.0                   0.0   
4                    23409.0                   0.0                   0.0   

                  Referee               Assistant 1  \
0  LOMBARDI Domingo (URU)     CRISTOPHE Henry (BEL)   
1       MACIAS Jose (ARG)  MATEUCCI Francisco (URU)   
2     TEJADA Anibal (URU)   VALLARINO Ricardo (URU)   
3   WARNKEN Alberto (CHI)       LANGENUS Jean (BEL)   
4     REGO Gilberto (BRA)      SAUCEDO Ulises (BOL)   

                  Assistant 2  RoundID  MatchID Home Team Initials  \
0         REGO Gilberto (BRA)    201.0   1096.0                FRA   
1       WARNKEN Alberto (CHI)    201.0   1090.0                USA   
2         BALWAY Thomas (FRA)    201.0   1093.0                YUG   
3    MATEUCCI Francisco (URU)    201.0   1098.0                ROU   
4  RADULESCU Constantin (ROU)    201.0   1085.0                ARG   

  Away Team Initials  
0                MEX  
1                BEL  
2                BRA  
3                PER  
4                FRA

In [62]:
### Find The shape of the Dataset
Data.shape

(4572, 20)

In [63]:
### sum of the duplicate rows in dataset
Data.duplicated().sum()

3735

In [64]:
### Remove duplicate rows

Data.drop_duplicates(inplace= True)

In [66]:
## Shape of the dataset after removing duplicates
Data.shape

(837, 20)

In [67]:
### NAN Values

Data.isnull().sum()

Year                    1
Datetime                1
Stage                   1
Stadium                 1
City                    1
Home Team Name          1
Home Team Goals         1
Away Team Goals         1
Away Team Name          1
Win conditions          1
Attendance              2
Half-time Home Goals    1
Half-time Away Goals    1
Referee                 1
Assistant 1             1
Assistant 2             1
RoundID                 1
MatchID                 1
Home Team Initials      1
Away Team Initials      1
dtype: int64

In [68]:
### look at the year feature NAN value row 
Data[Data['Win conditions'].isnull()]

Year Datetime Stage Stadium City Home Team Name  Home Team Goals  \
852   NaN      NaN   NaN     NaN  NaN            NaN              NaN   

     Away Team Goals Away Team Name Win conditions  Attendance  \
852              NaN            NaN            NaN         NaN   

     Half-time Home Goals  Half-time Away Goals Referee Assistant 1  \
852                   NaN                   NaN     NaN         NaN   

    Assistant 2  RoundID  MatchID Home Team Initials Away Team Initials  
852         NaN      NaN      NaN                NaN                NaN

In [69]:
### ### look at the Attendance feature NAN value row 
Data[Data['Attendance'].isnull()]

Year              Datetime        Stage            Stadium  \
823  2014.0  30 Jun 2014 - 17:00   Round of 16  Estadio Beira-Rio   
852     NaN                   NaN          NaN                NaN   

              City Home Team Name  Home Team Goals  Away Team Goals  \
823  Porto Alegre         Germany              2.0              1.0   
852            NaN            NaN              NaN              NaN   

    Away Team Name                 Win conditions  Attendance  \
823        Algeria  Germany win after extra time          NaN   
852            NaN                            NaN         NaN   

     Half-time Home Goals  Half-time Away Goals             Referee  \
823                   0.0                   0.0  RICCI Sandro (BRA)   
852                   NaN                   NaN                 NaN   

                   Assistant 1              Assistant 2   RoundID  \
823  DE CARVALHO Emerson (BRA)  VAN GASSE Marcelo (BRA)  255951.0   
852                        NaN                      NaN       NaN   

         MatchID Home Team Initials Away Team Initials  
823  300186460.0                GER                ALG  
852          NaN                NaN                NaN

### Observation 

1. Replace with the median in 823 row NAN VAlUE
2. Remove auther NAN values

In [70]:
Data.describe().T

count          mean           std     min      25%  \
Year                  836.0  1.984536e+03  2.229986e+01  1930.0   1970.0   
Home Team Goals       836.0  1.824163e+00  1.619178e+00     0.0      1.0   
Away Team Goals       836.0  1.021531e+00  1.072024e+00     0.0      0.0   
Attendance            835.0  4.485946e+04  2.355111e+04  2000.0  29800.0   
Half-time Home Goals  836.0  7.188995e-01  9.419952e-01     0.0      0.0   
Half-time Away Goals  836.0  4.270335e-01  6.750906e-01     0.0      0.0   
RoundID               836.0  1.086093e+07  2.751802e+07   201.0    262.0   
MatchID               836.0  5.677577e+07  1.070329e+08    25.0   1183.5   

                          50%          75%          max  
Year                   1990.0      2002.00       2014.0  
Home Team Goals           2.0         3.00         10.0  
Away Team Goals           1.0         2.00          7.0  
Attendance            41000.0     61009.00     173850.0  
Half-time Home Goals      0.0         1.00          6.0  
Half-time Away Goals      0.0         1.00          5.0  
RoundID                 337.0    249722.00   97410600.0  
MatchID                2113.5  43950047.25  300186515.0

In [79]:
### Drop 852th row 
Data.drop(852, inplace = True)


In [83]:
#### Replace Attendance feature NAN value with median
Data.fillna(Data['Attendance'].median(), inplace= True)

In [84]:
Data.isnull().sum()

Year                    0
Datetime                0
Stage                   0
Stadium                 0
City                    0
Home Team Name          0
Home Team Goals         0
Away Team Goals         0
Away Team Name          0
Win conditions          0
Attendance              0
Half-time Home Goals    0
Half-time Away Goals    0
Referee                 0
Assistant 1             0
Assistant 2             0
RoundID                 0
MatchID                 0
Home Team Initials      0
Away Team Initials      0
dtype: int64

In [86]:
Data.describe().T

count          mean           std     min      25%  \
Year                  836.0  1.984536e+03  2.229986e+01  1930.0   1970.0   
Home Team Goals       836.0  1.824163e+00  1.619178e+00     0.0      1.0   
Away Team Goals       836.0  1.021531e+00  1.072024e+00     0.0      0.0   
Attendance            836.0  4.485484e+04  2.353738e+04  2000.0  29800.0   
Half-time Home Goals  836.0  7.188995e-01  9.419952e-01     0.0      0.0   
Half-time Away Goals  836.0  4.270335e-01  6.750906e-01     0.0      0.0   
RoundID               836.0  1.086093e+07  2.751802e+07   201.0    262.0   
MatchID               836.0  5.677577e+07  1.070329e+08    25.0   1183.5   

                          50%          75%          max  
Year                   1990.0      2002.00       2014.0  
Home Team Goals           2.0         3.00         10.0  
Away Team Goals           1.0         2.00          7.0  
Attendance            41000.0     60984.50     173850.0  
Half-time Home Goals      0.0         1.00          6.0  
Half-time Away Goals      0.0         1.00          5.0  
RoundID                 337.0    249722.00   97410600.0  
MatchID                2113.5  43950047.25  300186515.0

In [91]:
### Features Datatype

Data.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 836 entries, 0 to 835
Data columns (total 20 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   Year                  836 non-null    float64
 1   Datetime              836 non-null    object 
 2   Stage                 836 non-null    object 
 3   Stadium               836 non-null    object 
 4   City                  836 non-null    object 
 5   Home Team Name        836 non-null    object 
 6   Home Team Goals       836 non-null    float64
 7   Away Team Goals       836 non-null    float64
 8   Away Team Name        836 non-null    object 
 9   Win conditions        836 non-null    object 
 10  Attendance            836 non-null    float64
 11  Half-time Home Goals  836 non-null    float64
 12  Half-time Away Goals  836 non-null    float64
 13  Referee               836 non-null    object 
 14  Assistant 1           836 non-null    object 
 15  Assistant 2           8

In [95]:
## Change Datetime feature datarype object to datetime

Data['Datetime'] = pd.to_datetime(Data['Datetime'])

In [96]:
Data.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 836 entries, 0 to 835
Data columns (total 20 columns):
 #   Column                Non-Null Count  Dtype         
---  ------                --------------  -----         
 0   Year                  836 non-null    float64       
 1   Datetime              836 non-null    datetime64[ns]
 2   Stage                 836 non-null    object        
 3   Stadium               836 non-null    object        
 4   City                  836 non-null    object        
 5   Home Team Name        836 non-null    object        
 6   Home Team Goals       836 non-null    float64       
 7   Away Team Goals       836 non-null    float64       
 8   Away Team Name        836 non-null    object        
 9   Win conditions        836 non-null    object        
 10  Attendance            836 non-null    float64       
 11  Half-time Home Goals  836 non-null    float64       
 12  Half-time Away Goals  836 non-null    float64       
 13  Referee             

In [97]:
Data.to_csv('cleaned_worldcupmatches.csv',index= False)